### PR TITLE
CV bugfix: handling multilinestrings

### LINKED ISSUE #1657: CV: TypeError: 'MultiLineString' object is not iterable
This is a somewhat rare edge case that comes up with complicated landmasses & AOIs. It's hard enough to contrive an example that we don't test for it, apparently. Our code to handle `MultLineString` iteration simply is not compatible with Shapely 2.0. https://shapely.readthedocs.io/en/stable/migration.html#multi-part-geometries-will-no-longer-be-sequences-length-iterable-indexable

I have some user's data so I can reproduce and fix it.

```python
Traceback (most recent call last):
  File "taskgraph\Task.py", line 674, in add_task
  File "taskgraph\Task.py", line 1093, in _call
  File "natcap\invest\coastal_vulnerability.py", line 1176, in prepare_landmass_line_index_and_interpolate_shore_points
TypeError: 'MultiLineString' object is not iterable
10/21/2024 17:29:44  natcap.invest.utils ERROR    Exception while executing natcap.invest.coastal_vulnerability
Traceback (most recent call last):
  File "natcap\invest\utils.py", line 165, in prepare_workspace
  File "invest\cli.py", line 470, in main
  File "natcap\invest\coastal_vulnerability.py", line 856, in execute
  File "taskgraph\Task.py", line 674, in add_task
  File "taskgraph\Task.py", line 1093, in _call
  File "natcap\invest\coastal_vulnerability.py", line 1176, in prepare_landmass_line_index_and_interpolate_shore_points
TypeError: 'MultiLineString' object is not iterable
```

### PR DESCRIPTION
**Pull request by davemfish**
Shapely 2.0 had some breaking changes, including how to iterate over multipart geometries. We missed one case in this model when we updated to shapely 2.0.

We already had a helper function to help with these cases. And I modified a test to cover it.

Fixes #1657 -- see for more details.

I also alphabetized the HISTORY, after leading with "General" and "Workbench" sections.

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the Workbench UI (if relevant)

### CODE REVIEW

File: tests/test_coastal_vulnerability.py [hunk 1]
```diff
@@ -1358,6 +1362,48 @@ def test_aoi_invalid_geometry(self):
                 aoi_path, landmass_path, model_resolution, target_vector_path,
                 polygon_pickle, lines_pickle, lines_rtree)
 
+    def test_prepare_landmass_invalid_geometry(self):
```
Comment (davemfish) [inline tests/test_coastal_vulnerability.py:1365]: This test I just moved from elsewhere in the file so it's close to other related tests.

File: tests/test_coastal_vulnerability.py [hunk 2]
```diff
@@ -1190,11 +1190,15 @@ def test_aoi_multiple_features(self):
...
+        # These two disjoint AOI polygons intersect the same landmass line
+        # segment. This tests an edge case where a MultiLineString
+        # geometry is created when landmass lines are clipped by the AOI.
         poly_a = Polygon([
             (-200, -200), (-100, -200), (-100, -100), (-200, -100),
             (-200, -200)])
         poly_b = Polygon([
-            (100, 100), (200, 100), (200, 200), (100, 200), (100, 100)])
+            (100, -200), (200, -200), (200, -100), (100, -100),
+            (100, -200)])
```
Comment (davemfish) [inline tests/test_coastal_vulnerability.py:1201]: This was the change to cover the bug.

**Review by emilyanndavis — APPROVED** [review]
